src/Raven.Server/Documents/Handlers/Processors/Documents/DocumentHandlerProcessorForGet.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Primitives;
+using Raven.Client;
+using Raven.Client.Documents.Operations.TimeSeries;
+using Raven.Client.Http;
+using Raven.Server.Documents.Includes;
+using Raven.Server.Documents.Queries.Revisions;
+using Raven.Server.Json;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Documents;
+
+internal class DocumentHandlerProcessorForGet : AbstractDocumentHandlerProcessorForGet<DocumentHandler, DocumentsOperationContext, Document>
+{
+    public DocumentHandlerProcessorForGet(HttpMethod method, [NotNull] DocumentHandler requestHandler) : base(method, requestHandler)
+    {
+    }
+
+    protected override bool SupportsShowingRequestInTrafficWatch => true;
+
+    protected override bool SupportsHandlingOfMissingIncludes => false;
+
+    protected override CancellationToken CancellationToken => RequestHandler.Database.DatabaseShutdown;
+
+    protected override void Initialize(DocumentsOperationContext context)
+    {
+        var txr = context.OpenReadTransaction();
```

Comment:
you need to open the tx only for the `DocumentHandlerProcessorForGet` so why not doing it explicitly ?

Reply:
Introduced `ExecuteInternalAsync` and wapped it by a read transaction in `DocumentHandlerProcessorForGet`